automation/release.sh

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+
+set -exuo pipefail
+
+function cleanup_gh_install() {
+    rm -f "gh_${GH_CLI_VERSION}_linux_amd64.tar.gz"
+    rm -rf "gh_${GH_CLI_VERSION}_linux_amd64/"
+}
+
+function ensure_gh_cli_installed() {
+    if command -V gh; then
+        return
+    fi
+
+    trap 'cleanup_gh_install' EXIT SIGINT SIGTERM
+
+    # install gh cli for uploading release artifacts, with prompt disabled to enforce non-interactive mode
+    curl -sSL "https://github.com/cli/cli/releases/download/v${GH_CLI_VERSION}/gh_${GH_CLI_VERSION}_linux_amd64.tar.gz" -o "gh_${GH_CLI_VERSION}_linux_amd64.tar.gz"
+    tar xvf "gh_${GH_CLI_VERSION}_linux_amd64.tar.gz"
+    sudo cp "gh_${GH_CLI_VERSION}_linux_amd64/bin/gh" /usr/local/bin/
```

Comment:
@dhiller I am not familiar with the environment on which this script will run, but can we assume that it will always allow us to run ```sudo``` and also w/o providing a password? I can think of another possible way to extract the ```gh``` package into a ```tmp``` directory and define ```_gh``` as a local wrapper that will refer to the ```tmp``` location in case of a successful fetch of the ```gh``` from GH site.

Reply:
That's a good point, I'll update on this.